src/store/sagas/pools.ts

```diff
@@ -77,11 +77,24 @@ export function* fetchAllPoolsForPairData(action: PayloadAction<PairTokens>) {
 }
 
 export function* fetchTicksAndTickMaps(action: PayloadAction<FetchTicksAndTickMaps>) {
-  const { tokenFrom, tokenTo, allPools } = action.payload
+  const { tokenFrom, tokenTo, poolKey } = action.payload
+  let poolsData = yield* select(poolsArraySortedByFees)
+
+  if (poolKey) {
+    let pools = poolsData.filter(pool => poolKeyToString(pool.poolKey) === poolKeyToString(poolKey))
+    let i = 0
+
+    while (pools.length === 0 && i < 20) {
+      yield* delay(50)
+      poolsData = yield* select(poolsArraySortedByFees)
+      pools = poolsData.filter(pool => poolKeyToString(pool.poolKey) === poolKeyToString(poolKey))
```

Comment:
maybe instead of waiting for pools it would be better to spawn and join a task that loads pools? also there is a case when 1000 ms is not enough and that something broke and task to load pools wasn't triggered